### PR TITLE
Youtube service

### PR DESCRIPTION
**Pull request by LarsVomMars**
Implementing the _YouTube API_ via [google-apis](https://github.com/googleapis/google-api-nodejs-client/blob/master/samples/youtube/upload.js).
~I currently have issues with the OAuth2 Flow, which fail due to _Invalid Request_ with missing authority.~

Creating OAuth2 Access like [this](https://developers.google.com/identity/protocols/oauth2/web-server#creatingcred).

~I would appreciate if someone can verify or disprove this issue - 'cos I'm kinda lost atm.~

### CODE REVIEW

File: nodecg-io-youtube/extension/index.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    async validateConfig(_config: YoutubeServiceConfig): Promise<Result<void>> {
+        return emptySuccess();
+    }
+
+    async createClient(config: YoutubeServiceConfig): Promise<Result<YoutubeServiceClient>> {
+        // https://developers.google.com/identity/protocols/oauth2/web-server#creatingcred
+        const auth = new google.auth.OAuth2({
+            clientId: config.clientID,
+            clientSecret: config.clientSecret,
+            redirectUri: "localhost:9090/nodecg-io-youtube/oauth2callback",
```
Comment (daniel0611) [inline nodecg-io-youtube/extension/index.ts:33]: @LarsVomMars we need to prepend the protocol here, in this case http. I've tested it with http and authentication worked. I didn't test any functions of the client though.
```suggestion
            redirectUri: "http://localhost:9090/nodecg-io-youtube/oauth2callback",
```

Comment (LarsVomMars) [reply nodecg-io-youtube/extension/index.ts:33]: lol

**Comment by LarsVomMars** [comment]
Seems to work<sup>TM</sup> (fine) now.

The whole process of getting it working is not that simple, so I will try to write a semi-complete and understandable doc.

**Comment by LarsVomMars** [comment]
If you don't have got any more suggestions, you can merge it now.

**Comment by daniel0611** [comment]
@LarsVomMars is it intented that the `/package-lock.json` has been removed in this PR?

**Comment by LarsVomMars** [comment]
It didn't get deleted, only changed a lot. Although I can't tell why this happened.

**Comment by daniel0611** [comment]
Oh, I see. Thanks for this @LarsVomMars

**Pull request merged**